name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/OnvistaPDFExtractor.java

```diff
@@ -1782,12 +1782,16 @@ private void addTaxVorabpauschaleTransaction()
                         .section("date").optional()
                         .match("^Nominal Ex-Tag Zahltag Jahreswert Vorabpauschale pro Stück")
                         // STK 1.212,000 02.01.2020 02.01.2020 EUR 0,1019
-                        .match("STK .* (?<exdate>\\d+\\.\\d+\\.\\d{4}+) (?<date>\\d+\\.\\d+\\.\\d{4}+) (\\w{3}+) .*")
+                        .match("STK .* (?<exdate>\\d+\\.\\d+\\.\\d{4}+) (?<date>\\d+\\.\\d+\\.\\d{4}+) (?<currency>\\w{3}+) .*")
                         .assign((t, v) -> {
+                            // if all taxes are covered by Freistellungauftrag/Verlustopf, section "Wert Konto-Nr. Betrag zu
+                            // Ihren Lasten" is not present -> extract currency here
+                            if (t.getCurrencyCode() == null)
+                                t.setCurrencyCode(asCurrencyCode(v.get("currency")));
                             t.setDateTime(asDate(v.get("date")));
                         })
 
-                        .wrap(t -> t.getAmount() != 0 ? new TransactionItem(t) : null);
+                        .wrap(TransactionItem::new);
```

Comment:
Won’t this create a transaction even if the amount is 0?

Reply:
Yes. The amount of 0.00 € paid taxes does exactly represent what is in the PDF (so there's nothing to debit from the reference account). My impression is that at least one transaction (even if it is somewhat empty) is required for the importer to recognise this as a proper Onvista file. I'm open for advise on how else to achieve that ;-)

Reply:
But what is the benefit to create a zero amount booking especially if they will never be editable via the PP interface? :thi

Reply:
Benefit is that it fixes the bug described in #1918 (failure to import PDF file)

Reply:
> My impression is that at least one transaction (even if it is somewhat empty) is required for the importer to recognise this as a proper Onvista file. I'm open for advise on how else to achieve that ;-)

If that’s currently the case, it could be changed by modifying the importer framework.

Reply:
Last year, I had a similar problem with the Consorsbank importer. For this, I created a ```Extractor.NonImportableItem```. This way, PP does not import a transactions that cannot be edited / used later on but at the same time the importer does not fail.

https://github.com/buchen/portfolio/blob/815a0d2f248f4faa1a5508775a6642c03b856ae8/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/ConsorsbankPDFExtractor.java#L543